fluentd/configs.d/filter-pre-mux-client.conf

```diff
@@ -5,6 +5,8 @@
   shared_key "#{File.open('/etc/fluent/muxkeys/shared_key') do |f| f.readline end.rstrip}"
   secure yes
   ca_cert_path /etc/fluent/muxkeys/ca
+  buffer_queue_limit "#{ENV['BUFFER_QUEUE_LIMIT']}"
```

Comment:
@nhosoi, the behavior of what to do with an exception is documented in the [stable version of fluentd, 0.12](http://docs.fluentd.org/v0.12/articles/buffer-plugin-overview), see description of `buffer_queue_full_action`.

Not too concerned about the performance, but more concerned about lost messages if fluentd is restarted and unable to flush it buffers when using the `memory` buffer mechanism.

I don't think we should make the buffering type configurable unless we get demand for it, so switching to `file` buffer should be considered a permanent switch for now.

Certainly tracking future work tuning the number of threads for fluentd behavior of the `out_forward`, `out_secure_forward`, and the `out_elasticsearch` (see [Buffered Output Options](https://github.com/uken/fluent-plugin-elasticsearch#buffered-output-options)) output plugins would be great.